runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -102,9 +102,10 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
       end_lnum = _end.line,
       end_col = line_byte_from_position(buf_lines, _end.line, _end.character, offset_encoding),
       severity = severity_lsp_to_vim(diagnostic.severity),
-      message = diagnostic.message,
+      message = table.concat(vim.lsp.util.split_lines(diagnostic.message), '\n'),
```

Comment:
Should the lines be normalized here, in vim.lsp.diagnostic, or handled by vim.diagnostic according to LSP rules?

Reply:
This is just the diagnostic message displayed, I don't think we need to split and concatenate here? Is the server actually returning anything other than `\n`?

Reply:
> This is just the diagnostic message displayed, I don't think we need to split and concatenate here?

https://github.com/neovim/neovim/blob/9d6a475cedf93a03214ca9c9bd0673a763994d78/runtime/lua/vim/diagnostic.lua#L1230

> Is the server actually returning anything other than `\n`?

Unclear, the specification doesn't state anything concrete either